.github/workflows/dependencies.yml

```diff
@@ -15,24 +15,34 @@ jobs:
       fail-fast: false
       matrix:
         target:
-          - repository: status-im/nimbus-eth2
+          - id: nimbus
+            repository: status-im/nimbus-eth2
             ref: unstable
-            token: ${{ secrets.ACTIONS_GITHUB_TOKEN_NIMBUS_ETH2 }}
-          - repository: waku-org/nwaku
+          - id: nwaku
+            repository: waku-org/nwaku
             ref: master
-            token: ${{ secrets.ACTIONS_GITHUB_TOKEN_NWAKU }}
-          - repository: codex-storage/nim-codex
+          - id: codex
+            repository: codex-storage/nim-codex
             ref: master
-            token: ${{ secrets.ACTIONS_GITHUB_TOKEN_NIM_CODEX }}
     steps:
+      - name: Set secret based on matrix target
+        run: |
+          if [ "${{ matrix.target.id }}" = "nimbus" ]; then
+            echo "REPO_TOKEN=${{ secrets.ACTIONS_GITHUB_TOKEN_NIMBUS_ETH2 }}" >> $GITHUB_ENV
+          elif [ "${{ matrix.target.id }}" = "nwaku" ]; then
+            echo "REPO_TOKEN=${{ secrets.ACTIONS_GITHUB_TOKEN_NWAKU }}" >> $GITHUB_ENV
+          elif [ "${{ matrix.target.id }}" = "codex" ]; then
+            echo "REPO_TOKEN=${{ secrets.ACTIONS_GITHUB_TOKEN_NIM_CODEX }}" >> $GITHUB_ENV
+          fi
```

Comment:
I think it should be possible to simplify this a bit:

In the targets you can add the name of the secret variable like this
```
- id: nimbus
  repository: status-im/nimbus-eth2
  ref: unstable
  secret: ACTIONS_GITHUB_TOKEN_NIMBUS_ETH2      <----------
  ```
  
  and then in the `run`:
  ```
echo "REPO_TOKEN=${{ secrets[matrix.target.secret] }} >> $GITHUB_ENV
```

Reply:
yep, beautifully suggestion! thnx